File: XOutput.Api/Rest/Input/InputDeviceInfo.cs

using System.Collections.Generic;
using XOutput.Common.Input;

namespace XOutput.Rest.Input
{
    public class InputDeviceInfo
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string DeviceApi { get; set; }
        public IEnumerable<InputDeviceSource> Sources { get; set; }
        public IEnumerable<InputDeviceTarget> Targets { get; set; }
    }
}

File: @xoutput/api/src/rest/input/InputDeviceInfo.ts
import { InputDeviceSource } from '../../common/input/InputDeviceSource';
import { InputDeviceTarget } from '../../common/input/InputDeviceTarget';

export type InputDeviceInfo = {
  id: string;
  name: string;
  deviceApi: string;
  sources: InputDeviceSource[];
  targets: InputDeviceTarget[];
};
